blog: Adds blog post documenting the `FOSS in Science devroom` at IndiaFOSS 2025 (#36)
scipy-india/scipy-india.github.io: blogs/community-calls/indiafoss-2025-science-devroom.md

# IndiaFOSS 2025 | Devroom – FOSS in Science

_Srihari Thyagarajan_

This blog post is our account of SciPy India's participation in the FOSS in Science devroom at IndiaFOSS 2025, held in Bengaluru. Here, we describe the devroom as it happened, including the talks, discussions, and the incredible diversity of applications of free and open-source software in scientific research and practice across India.

## Details

> - **Date**: September 21th, 2025

Reviewer: ```suggestion
- **Date**: September 21st, 2025
```

- **Time**: 10:00 to 13:00 IST
- **Venue**: [NIMHANS Convention Centre, Bengaluru](https://osmapp.org/way/1219285692#15.37/12.9430/77.5956)

> - **Platform**: In-person with livestream on [YouTube](https://www.youtube.com/watch?v=NyQced4oJVM)

Reviewer: I suppose the venue being listed above this line implies that the event took place in person, correct? We could reword this:

```suggestion
- **YouTube livestream**: https://www.youtube.com/watch?v=NyQced4oJVM
```

## Background

> IndiaFOSS 2025 brought together the Indian FOSS community for two days of talks, workshops, and networking. Among the [various devrooms](https://fossunited.org/indiafoss/devrooms), the FOSS in Science track carved out a space for researchers, practitioners, and enthusiasts to discuss how open-source software is _transforming_ scientific work — from data acquisition in STEM education to formalizing mathematical proofs, from tracking bird populations to building digital public infrastructure for landscape planning.

Reviewer: ```suggestion
IndiaFOSS 2025 brought together the Indian FOSS community for two days of talks, workshops, and networking. Among the [various devrooms](https://fossunited.org/indiafoss/devrooms), the [FOSS in Science devroom](https://fossunited.org/indiafoss/2025/devrooms/science) carved out a space for researchers, practitioners, and enthusiasts to discuss how open-source software is _transforming_ scientific work — topics included data acquisition in STEM education, formalizing mathematical proofs, tracking bird populations, and building digital public infrastructure for landscape planning.
```

## Closing remarks

After a full day of talks spanning hardware hacking, infrastructure design, data analysis, policy, and everything in between, the devroom concluded with announcements about the [SciPy India community](https://scipy-india.github.io/). We shared information about our [first community call](https://scipy-india.github.io/blogs.html?id=community-call-july-2025) in July, our upcoming plans for more community calls and workshops, and our hopes for an _in-person_ conference later this year or early next year. We encouraged attendees to join our [Zulip chat](https://scipyindia.zulipchat.com/join/4mesdxfbbpl4titgtdzx4iwv/), check out our open-source repositories, and fill out the feedback form with their thoughts on the community and preferences for future events.

## Livestream

> The full [FOSS in Science devroom](https://fossunited.org/indiafoss/2025/devrooms/science) was livestreamed on [YouTube](https://www.youtube.com/watch?v=NyQced4oJVM) for those who couldn't attend in person. The recording remains available for viewing at your convenience, with timestamps for individual talks.

Reviewer: I wanted to mention that the full playlist is now available: https://www.youtube.com/playlist?list=PLOGilj110oly_VPJ7pz0avjyxiLjiTJTP. Perhaps it might make sense, after applying the above suggestions, to include a "Videos" heading and a Markdown table with the speakers and their video links?

Author: Ah sorry, this review slipped my eye. Adding a `## Videos` section with a markdown table pointing to respective talks (CFPs and timestamped section/chapter in the livestream).
That should work.

## In a nutshell

Thank you to everyone who spoke, attended, asked questions, and contributed to making the FOSS in Science devroom at IndiaFOSS 2025 a success. To the speakers who traveled to Bengaluru and prepared thoughtful presentations showcasing their work: your contributions made the day what it was. To FOSS United for organizing IndiaFOSS and providing us with the platform and infrastructure to host the devroom: we're grateful for your continued support of the FOSS community in India.

> We look forward to seeing you at future community events, whether online or in person! To stay updated, join our [Zulip chat](https://scipyindia.zulipchat.com/join/4mesdxfbbpl4titgtdzx4iwv/) and follow us on our [social media channels](https://scipy-india.github.io/contact.html). And to you, the reader, thank you for taking the time to read about our day in Bengaluru. We hope it inspires you to explore, contribute, and connect with the FOSS in science community.

Reviewer: ```suggestion
We look forward to seeing you at future community events, whether online or in person! To stay updated, join our [Zulip chat](https://scipyindia.zulipchat.com/join/4mesdxfbbpl4titgtdzx4iwv/) and follow us on our [social media channels](https://scipy-india.github.io/contact.html). And to you, the reader, thank you for taking the time to read about our day in Bengaluru. We hope it inspires you to explore, contribute, and connect with the scientific FOSS community.
```

"FOSS in science community" sounds a bit odd, so take it or leave it :)

Pradyot Ranjan's presentation on [the Array API Standard](https://fossunited.org/c/indiafoss/2025/cfp/4vqtvphjlo) made a convincing argument for why the widespread use of array libraries — NumPy, CuPy, PyTorch, JAX, and others — requires a standardised API specification. The standard aims to provide consistent APIs and array operations across different Python libraries, enabling backend-agnostic code that can run on CPUs, GPUs, or specialized hardware without modification. This is the kind of infrastructure work that doesn't make headlines but makes everything else possible: when consumer libraries like scikit-learn can write against the Array API instead of assuming NumPy, the entire ecosystem becomes more interoperable and sustainable.

> Sanket Verma, a long-time maintainer of [`Zarr`](https://github.com/zarr-developers/zarr-python) and a Technical and Administrative Board member of the NumFOCUS committee, presented [what's new with Zarr](https://fossunited.org/c/indiafoss/2025/cfp/840daar96d), the cloud-optimized storage format for n-dimensional arrays. Created initially by Alistair Miles in 2016 for handling massive genomic datasets from malaria research, Zarr's approach of dividing large arrays into compressed chunks that can be stored and loaded selectively has proven transformative for scientific computing at scale. The recently released Zarr Python 3 brings a leaner specification, improved handling of high-latency storage, and extensibility through entry point mechanisms for custom stores and codecs. With institutions using Zarr to store datasets approaching 50 petabytes in the cloud, Sanket's update on the active community — weekly meetings, office hours, and a welcoming environment for new contributors — reinforced that this is infrastructure built for the long haul.

Reviewer 2: ```suggestion
Sanket Verma, a long-time maintainer of [`Zarr`](https://github.com/zarr-developers/zarr-python) and a Technical and Administrative Board member of the NumFOCUS, presented [what's new with Zarr](https://fossunited.org/c/indiafoss/2025/cfp/840daar96d), the cloud-optimized storage format for N-dimensional arrays. Created initially by Alistair Miles in 2016 for handling massive genomic datasets from malaria research, Zarr's approach of dividing large arrays into compressed chunks that can be stored and loaded selectively has proven transformative for scientific computing at scale. The recently released Zarr Python 3 brings a leaner specification, improved handling of high-latency storage, and extensibility through entry point mechanisms for custom stores and codecs. With institutions using Zarr to store datasets approaching petabytes in the cloud, Sanket's update on the active community — weekly meetings, office hours, and a welcoming environment for new contributors — reinforced that this is infrastructure built for the long haul.
```
